src/overview.md

```diff
@@ -0,0 +1,322 @@
+# Rust compiler Overview
+
+This chapter is about the overall process of compiling a program -- how
+everything fits together.
+
+The rust compiler is special in two ways: it does things to your code that
+other compilers don't do (e.g. borrow checking) and it has a lot of
+unconventional implementation choices (e.g. queries). We will talk about these
+in turn in this chapter, and in the rest of the guide, we will look at all the
+individual pieces in more detail.
+
+## What the compiler does to your code
+
+So first, let's look at what the compiler does to your code. For now, we will
+avoid mentioning how the compiler implements these steps except as needed;
+we'll talk about that later.
+
+**TODO: Would be great to have a diagram of this once we nail down the details...**
+
+**TODO: someone else should confirm this vvv**
+
+- The compile process begins when a user writes a Rust source program in text and invokes the `rustc` compiler on it. The work that the compiler needs to perform is defined by command-line options. For example, it is possible to enable nightly features (`-Z` flags), perform `check`-only builds, or emit LLVM-IR rather than executable machine code. The `rustc` executable call may be indirect through the use of `cargo`.
```

Comment:
https://github.com/rust-lang/rustc-dev-guide/pull/659#discussion_r402821994

> Should we have a soft wrap here?

> Open question how we will format the entire document. Maybe this will become standard body copy with paragraphs rather than a bullet list?

Reply:
I think if we are going to do paragraphs, we should make them actually paragraphs, rather than bullets. But if we are going to do bullets, we should keep the, short.

Personally I'm ok with doing paragraphs.

Reply:
> Personally I'm ok with doing paragraphs.

I like that idea too.  I think that we may be hitting lengths that are a bit too much for lists.

Reply:
One thing, though is that we should resolve all of the above comments first. Otherwise, github will mark them all as outdated.

Reply:
Will do this in a followup ( #320)